staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -373,7 +373,7 @@ func (d *DiscoveryClient) SwaggerSchema(version schema.GroupVersion) (*swagger.A
 
 // OpenAPISchema fetches the open api schema using a rest client and parses the proto.
 func (d *DiscoveryClient) OpenAPISchema() (*openapi_v2.Document, error) {
-	data, err := d.restClient.Get().AbsPath("/swagger-2.0.0.pb-v1").Do().Raw()
+	data, err := d.restClient.Get().SetHeader("Cache-control", "max-stale=0").AbsPath("/swagger-2.0.0.pb-v1").Do().Raw()
```

Comment:
This is a hack to explicitly make caching opt-in and only for openapi json in 1.8.  We need the server to start sending the cache-control headers in the responses to generally open up caching to resources, since the client doesn't know what is acceptable to cache, and what is not.  Once we start caching resources, we will remove this opt-in enforcement mechanism, and default to what the server tells us to do.